doc/options.md

```diff
@@ -0,0 +1,2865 @@
+# Options Reference 
+
+1. [`inputs.iogx.lib.mkFlake`](#mkflake) 
+    Makes the final flake outputs.
+2. [`pkgs.lib.iogx.mkProject`](#mkproject) 
+    Makes a [`haskell.nix`](https://github.com/input-output-hk/haskell.nix) project decorated with the `iogx` overlay.
+3. [`pkgs.lib.iogx.mkShell`](#mkshell)
+    Makes a `devShell` with `pre-commit-check` and tools.
+
+---
+
+### `mkFlake`
+
+**Type**: core API function
+
+
+
+**Example**: 
+```nix
+# flake.nix
+{
+  outputs = inputs: inputs.iogx.lib.mkFlake {
+    inherit inputs;
+    repoRoot = ./.;
+    debug = false;
+    nixpkgsArgs = {};
+    systems = [ "x86_64-linux" "x86_64-darwin" ];
+    outputs = { repoRoot, inputs, pkgs, lib, system }: [];
+  };
+}
+
+```
+
+
+The `inputs.iogx.lib.mkFlake` function takes an attrset of options and returns an attrset of flake outputs.
+
+In this document, options for the input attrset are prefixed by `mkFlake.<in>`.
+
+
+---
+
+### `mkFlake.<in>.debug`
+
+**Type**: boolean
+
+**Default**: `false`
+
+
+
+
+If enabled, IOGX will trace debugging info to standard output.
+
+
+---
+
+### `mkFlake.<in>.flake`
+
+**Type**: attribute set
+
+**Default**: `{ }`
+
+
+**Example**: 
+```nix
+{
+  lib.bar = _: null;
+
+  packages.x86_64-linux.foo = null;
+  devShells.x86_64-darwin.bar = null;
+
+  networks = {
+    prod = { };
+    dev = { };
+  };
+}
+
+```
+
+
+A flake-like attrset.
+
+You can place additional flake outputs here, which will be recursively updated with the attrset from [`mkFlake.<in>.outputs`](#mkflakeinoutputs).
```

Comment:
We should link to something describing the merging algorithm here, as "recursively updated" is not a trivial problem.